pylint/testutils/checker_test_case.py

```diff
@@ -53,10 +59,15 @@ def assertAddsMessages(self, *messages: MessageTest) -> Generator[None, None, No
 
         for expected_msg, gotten_msg in zip(messages, got):
             assert expected_msg.msg_id == gotten_msg.msg_id, msg
-            assert expected_msg.line == gotten_msg.line, msg
             assert expected_msg.node == gotten_msg.node, msg
             assert expected_msg.args == gotten_msg.args, msg
             assert expected_msg.confidence == gotten_msg.confidence, msg
+
+            if ignore_position:
```

Comment:
Is it possible for us to only check `line` and `col_offset` and then skip the other two when `ignore_position` is given?
I think we need at least a `line` and `col_offset` for messages to be displayable in IDEs.

Reply:
line and col_offset should be calculated from the given node.
The `ignore_position` is intended to create unit test in the sense of "Check only this unit/function", so this should not depend on the functionality of extracting the nodes/col.

Also this makes the unit test much smaller/easier to read.